NoiseEngine/Interop/InteropArray.cs

```diff
@@ -130,7 +131,7 @@ public struct InteropArray<T> : IDisposable, IReadOnlyList<T> where T : unmanage
     /// Disposes this object.
     /// </summary>
     public unsafe void Dispose() {
-        if (pointer == null)
+        if (pointer == null || MemoryHelper.IsDangling<T>((nuint)pointer) || Length <= 0)
```

Comment:
This should be `pointer == null || (Length == 0 && IsDangling())`, because you should check this only where Length is zero.

Reply:
Yep changed that, my bad.

Reply:
You change it wrong. See how I wrote again.